src/Symfony/Component/Yaml/Parser.php

```diff
@@ -286,7 +286,11 @@ public function parse($value, $exceptionOnInvalidType = false, $objectSupport =
             }
 
             if ($isRef) {
-                $this->refs[$isRef] = end($data);
+                if (!isset($key)) {
```

Comment:
and another solution would be to move this ref assignment inside the block corresponding to mappings and sequences instead of using a common block at the end